mypy/semanal.py

```diff
@@ -2128,6 +2154,7 @@ def add_field(var: Var, is_initialized_in_class: bool = False,
         add_field(Var('_field_types', dictype), is_initialized_in_class=True)
         add_field(Var('_field_defaults', dictype), is_initialized_in_class=True)
         add_field(Var('_source', strtype), is_initialized_in_class=True)
+        add_field(Var('__annotations__', ordereddictype), is_initialized_in_class=True)
```

Comment:
This could be unrelated to this PR, but despite the fact that you define it here as ``ordereddicttype`` it is revealed as ``dict`` in tests. Also, ``__annotations__`` is still just a ``dict`` (although it is ordered in 3.6+) and does not have ``OrderedDict`` methods like ``move_to_end()``.

Also I am not sure why you need it here. Latest typeshed stubs define ``object.__annotations__``